File: punkwm_dock/src/services/panel/apps.js
class PanelApps extends Sender {
    static async Open(){super.Emit("Panel:Apps", "Open");}
    static async LoadApps(){super.Emit("Panel:Apps", "Load Apps");}
    static async OpenApp(package){super.Emit("Panel:Apps", "Open App", {"package": package});}
    static async SearchApps(query){super.Emit("Panel:Apps", "Search", {"q": query });}
}

File: com_punkwm_dock/src-tauri/src/ipc/dock/handlers/panel/apps.rs
use crate::ipc::socket::socket_send;
use crate::IpcMessage;
use serde_json::json;
use crate::ipc_front::message::IpcFrontMessage;


pub fn ipc_front_handler_apps_panel(msg: IpcFrontMessage) {
     match msg.get_name().as_str() {
        "Open" =>{
            let command = IpcMessage::new("Panel:Apps", "Open", json!({}));
            let _ = socket_send(&command);
        },
        
        "Search" =>{
            let command = IpcMessage::new("Panel:Apps", "Search", msg.get_data());
            let _ = socket_send(&command);
        },

        "Load Apps" =>{
            let command = IpcMessage::new("Panel:Apps", "Load Apps", json!({}));
            let _ = socket_send(&command);
        },

        "Open App" =>{
            let command = IpcMessage::new("Panel:Apps", "Open App", msg.get_data());
            let _ = socket_send(&command);
        },

        _ => {
            println!(
                "Nombre desconocido: [{}:{}]",
                msg.get_category(),
                msg.get_name()
            );
        }
     }
    
}



